providers/ldap/ad/ad_provider.go

```diff
@@ -63,6 +64,11 @@ func InitializeProvider() *ADProvider {
 	ldapClient := &ldap.LClient{}
 	ldapProvider := &ADProvider{}
 
+	pool, err := x509.SystemCertPool()
+	if err != nil {
+		log.Fatal("Could not load certs")
```

Comment:
Oh shoot: I didn't even realize this was fatal. I just meant that you should add the `err` to the log statement so that we know what the error was. But now that you point out that this is Fatal log statement, don't do that. Return an error to a top level.